gbus/saga/glue.go

```diff
@@ -103,6 +104,9 @@ func (imsm *Glue) getDefsForMsgName(msgName string) []*Def {
 
 func (imsm *Glue) handleNewSaga(def *Def, invocation gbus.Invocation, message *gbus.BusMessage) error {
 	newInstance := def.newInstance()
+	if message.SagaCorrelationID != "" {
```

Comment:
this is tricky code and works because of context setup in the calling function which makes it difficult to reason about so please add commentary explaining what and why in more detail or refactor to a more straight forward  approach